homeassistant/components/prosegur/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""The Prosegur Alarm integration."""
+import asyncio
+
+from pyprosegur.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_COUNTRY, DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Prosegur Alarm component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Prosegur Alarm from a config entry."""
+    session = aiohttp_client.async_get_clientsession(hass)
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = Auth(
+        session,
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        entry.data[CONF_COUNTRY],
+    )
+
```

Comment:
Maybe it's a good idea to run validate_input here as well. The credentials being stored in the config entry may be outdated or otherwise not functioning. Checking it here and returning false notifies home assistant that the integration failed to setup.